### PR TITLE
Listen timeout when used with Promise.resolve()

### PR DESCRIPTION
**Pull request by allevo**
This PR adds a test to show the problem opened by https://github.com/fastify/fastify/issues/165 but not exampled.

travis should fail

### CODE REVIEW

**Comment by mcollina** [comment]
Wow, this is a very insteresting one.

Have you found out if it's a bug here or in avvio?

Is it happening even with a registered plugin?

**Comment by allevo** [comment]
> Wow, this is a very insteresting one.

Yep! Before opening the linked issue, I've spent 2h!

> Have you found out if it's a bug here or in avvio?

Deeping into [fastify](https://github.com/fastify/fastify/blob/master/fastify.js#L189) I think this can be reported to an `avvio` issue.

> Is it happening even with a registered plugin?

I don't know. This evening (CEST) I'll check

**Comment by mcollina** [comment]
@allevo open up an issue on `avvio`  as well if you can reproduce it there.

**Comment by mcollina** [comment]
The test is still failing, even if I fixed the bug in avvio. Maybe there is something else?

File: test/listen.test.js [hunk 1]
```diff
@@ -20,3 +21,16 @@ test('listen accepts a port, address, and callback', t => {
...
   })
 })
+
+test('listen after Promise.resolve()', t => {
+  t.plan(2)
+  const f = Fastify()
+  Promise.resolve()
+    .then(() => {
+      f.listen(0, (err) => {
+        fastify.server.unref()
```
Comment (delvedor) [inline test/listen.test.js]: This should be just `f`, otherwise the process will hang.

Comment (allevo) [inline test/listen.test.js]: changed

**Comment by mcollina** [comment]
Thanks @allevo!!!

**Pull request merged**